pkg/kp/rcstore/store.go

```diff
@@ -31,7 +31,13 @@ type Store interface {
 	//
 	// The lock will be taken against an ephemeral key, so it is safe to use a
 	// session that deletes its keys on invalidation.
-	Lock(id fields.ID, session string) (bool, error)
+	//
+	// The separate locks are for one reader (who watches the RC and acts when
+	// it changes) and one writer (who mutates the RC to cause those changes).
+	// Each lock may be held by one, and only one, party (readers cannot share
+	// the read lock).
```

Comment:
Why isn't that just a function of a separate locking mechanism over the preparers themselves? It seems orthogonal to the task of locking an RC itself.